internal/envtest/multi_envtest.go

```diff
@@ -0,0 +1,26 @@
+package envtest
+
+import (
+	"testing"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type MultiK8sEnvironment []*Environment
+
+func RunMulti(m *testing.M, setupFunc func(e *MultiK8sEnvironment), numClusters int) (code int) {
+	e := make(MultiK8sEnvironment, numClusters)
+	ctx := ctrl.SetupSignalHandler()
```

Comment:
suggestion: You probably want to do WithCancel here and have a cancel function to do the stopping.

Reply:
You're right, that's how I've done this in k8ssandra-operator's tests. I'll do it here too.

Reply:
I tried to implement this:

```
func RunMulti(m *testing.M, setupFunc func(e *MultiK8sEnvironment), numClusters int) (code int) {
	e := make(MultiK8sEnvironment, numClusters)
	ctx := ctrl.SetupSignalHandler()
	ctx, cancel := context.WithCancel(ctx)
	for i := 0; i < numClusters; i++ {
		e[i] = NewEnvironment(ctx)
		e[i].cancelManager = cancel
		e[i].Start()
	}
	defer func() {
		for i := 0; i < numClusters; i++ {
			e[i].Stop()
		}
	}()
	setupFunc(&e)
	exitCode := m.Run()
	return exitCode
}
```

but it fails linting for some reason with `the cancel function is not used on all paths (possible context leak)`. I don't believe this error, but what do you think @burmanm ?

Reply:
Go Vet issue means we can't improve on the current solution, we add a cancel manager elsewhere anyway.